Here you the main scripts written for "De novo insight into human de novo genes"

Only the scripts are presents in the folder. If you need to access the data, please go in the supplemental data of the paper, or directly contact the authors




*******************************************************************************************************************************
Create correspondance file for de novo genes, genomic positoin and category
*******************************************************************************************************************************

Pour ca je combine 3 fichiers qui sont dans le directory avec un code que j ai creer

Path Motif/DeNovoData

--------------------------------
python MakeFinalCorrespondanceListe.py
--------------------------------

------->>>>>>>> CorrespondanceFinal1


*******************************************************************************************************************************
Add transcription direction (Forward or Reverse)
*******************************************************************************************************************************


Path Motif/humanGeneralData/ChromosomesFasta



--------------------------------
python GetDeNovoStrend.py
--------------------------------



------->>>>>>>> CorrespondanceFinal2 



*******************************************************************************************************************************
Retrieve siz 5'and 3'UTRs of de novo genes
*******************************************************************************************************************************


Path Motif/DeNovoData/RecupDebutTransc


--------------------------------
python C1_GetTranscriptLen.py
--------------------------------


------->>>>>>>> FileContigsSize



--------------------------------
python C2_Get5PrimeforwardTranscripts.py
--------------------------------


------->>>>>>>> ListeForward5PrimeSize
------->>>>>>>> ListeReverse5PrimeSize



--------------------------------
python C3_GetReverseInfo.py
--------------------------------


------->>>>>>>> ORF_5_3_prime_size



--------------------------------
python C4_BuildCorres3.py
--------------------------------


------->>>>>>>> CorrespondanceFinal3


*******************************************************************************************************************************
Get sequence 200 bp before transcription starting point of de novo genes
*******************************************************************************************************************************


PathMotif/humanGeneralData/ChromosomesFasta



--------------------------------
python GetDeNovo200.py
--------------------------------


------->>>>>>>> Sequences200AfterDeNovoGene
------->>>>>>>> Sequences200BeforeDeNovoGene



------->>>>>>>> Nom des doublons (des ORFS provanant du meme transcript
------->>>>>>>> SpecialCorres200

Same between -100bp and +100bp:

--------------------------------
python GetDeNovo200BetweenProm.py
--------------------------------

*******************************************************************************************************************************
Get size and number of de novo genes per chromosomes 
*******************************************************************************************************************************


Path Motif/DeNovoData


--------------------------------
python NbDeNovoAndSize.py
--------------------------------

------->>>>>>>> DeNovo_Size_Chrom: Les tailles pour les 27491 de novo genes, 27550 de novo genes, moins 59 qui sont sur d autres chromosomes, 11 mito et 48 unknown
------->>>>>>>> DeNovo_Special_Size_Chrom: Les tailles pour les 23135 de novos genes bons




*******************************************************************************************************************************
Get nucleotidic seq of 27491 de novo genes
*******************************************************************************************************************************

path Motif/DeNovoData
--------------------------------
python RetrieveGoodORF.py
--------------------------------


------->>>>>>>> DeNovoNucFinal.fa





*******************************************************
*********************************************************************
************************************************************************************

 ++++++++++++    Human genes, NcRNA et pseudogenes  +++++++++++++++

************************************************************************************
*********************************************************************
*******************************************************






*******************************************************************************************************************************
create list of genomic positions for genes, pseudogenes and NcRNA
*******************************************************************************************************************************


Path Motif/humanGeneralData/GenesNcPseudoPos


--------------------------------
python ExtractGenesFromGff.py
--------------------------------



------->>>>>>>>  AllGenesPos

Total

21445 genes
23934 RnRNA
15200 pseudogenes


*******************************************************************************************************************************
Get list sequences 200 bp upstream genes, pseudogenes et NcRNA
*******************************************************************************************************************************



Path Motif/humanGeneralData/ChromosomesFasta

--------------------------------
GetGenesEtc200.py
--------------------------------




------->>>>>>>> Genes200After
------->>>>>>>> Genes200Before
------->>>>>>>> NcRNAgenes200After
------->>>>>>>> NcRNAgenes200Before
------->>>>>>>> Pseudogenes200After
------->>>>>>>> Pseudogenes200Before


Same for -100 to +100 between transcription starting point

*******************************************************************************************************************************
Retrieve humain genes and NcRNA sequences
*******************************************************************************************************************************

Downloaded from ensembl

Homo_sapiens.GRCh38.cds.all.fa, All human transcripts
Homo_sapiens.GRCh38.ncrna.fa, All human NcRNA


CDS file: 113656 transcripts
NcRNA file: 58313 transcritps


Get longest transcript per gene/NcRNA


--------------------------------
python GetLongestTranscript.py
--------------------------------


------->>>>>>>> GenesLongestTranscripts
------->>>>>>>> NcRNALongestTranscripts


GenesLongestTranscripts: 23483 sequences
NcRNALongestTranscripts: 25537 sequences


Revove doublons


--------------------------------
python RetrieveSeqsFromGenesPos.py
--------------------------------


------->>>>>>>> GenesLongestTranscriptFinal
------->>>>>>>> NcRNALongestTranscriptFinal

GenesLongestTranscriptFinal: 21401 genes
NcRNALongestTranscriptFinal: 23958



*******************************************************
*********************************************************************
************************************************************************************

 ++++++++++++    Introns et Non coding regions  +++++++++++++++

************************************************************************************
*********************************************************************
*******************************************************







*******************************************************************************************************************************
Remove unusual chromosomes
*******************************************************************************************************************************



------->>>>>>>> IntronsPositionsWithoutEnd



*******************************************************************************************************************************
Create list non coding positions
*******************************************************************************************************************************



--------------------------------
python BuildGenesPLusDeNovo.py
--------------------------------

------->>>>>>>> AllGenesPosANDdeNovo



--------------------------------
python GetNcRegionsBon.py
--------------------------------



------->>>>>>>> NonCodingRegionsPositions




*******************************************************************************************************************************
Create FASTA file with all non coding sequences and introns
*******************************************************************************************************************************


path Motif/humanGeneralData/ChromosomesFasta

--------------------------------
python ChromoSize_GetNc.py
--------------------------------



------->>>>>>>> Intron_chr1 ***
------->>>>>>>> NonCoding_chr1 ***



*******************************************************************************************************************************
Randomly draw sequences
*******************************************************************************************************************************



--------------------------------
python RandomNonCoding.py
--------------------------------

------->>>>>>>> RandomNonCodingSeqs.fa



*******************************************************************************************************************************
Sme for introns
*******************************************************************************************************************************



--------------------------------
RandomIntron.py
qsub RunPython.sh
--------------------------------


------->>>>>>>> RandomIntronicSeqs.fa


*******************************************************************************************************************************
Get 200 nucleotidesin each non coding region randomly selected
*******************************************************************************************************************************



--------------------------------
python Take_seq200_random.py
--------------------------------

------->>>>>>>> NonCoding200Random.fa


*******************************************************************************************************************************
Get 200 nucleotidesin each intron randomly selected
*******************************************************************************************************************************



------->>>>>>>> Intron200Random.fa



*******************************************************************************************************************************
Search for Domains in DeNovo, Genes, NonCoding, Intronic, NcRNA
*******************************************************************************************************************************



Path DataForDomains/



--------------------------------
python GetProt.py
--------------------------------

------->>>>>>>> DeNovoGenesProt.fa
------->>>>>>>> HumanGenesProt.fa





*******************************************************************************************************************************
Domains Analyse
*******************************************************************************************************************************



Path: Motif/DomainAnalyse

--------------------------------
python GetDifferentOutputs.py
--------------------------------

------->>>>>>>> DataDomains.rst


*******************************************************************************************************************************
5'and 3'UTR
*******************************************************************************************************************************


path Motif/Data5_3_UTR


---------------------------------
python Extract5_3_FromGff.py (a faire tourner sur mon ordi)
--------------------------------

------->>>>>>>> DataHumanGenes_5_3_prime


--------------------------------
python ExtractDataForDeNovo.py
--------------------------------

------->>>>>>>> DeNovo_5_3prime

Path Motif/humanGeneralData/ChromosomesFasta



--------------------------------
python Get5_3_primeRegions.py
--------------------------------

------->>>>>>>> Region3primeDeNovoGenes
------->>>>>>>> Region3primeGenes
------->>>>>>>> Region5primeDeNovoGenes
------->>>>>>>> Region5primeGenes


*******************************************************************************************************************************
Search for motifs
*******************************************************************************************************************************

Path Motif/MotifsSearch



--------------------------------
qsub RunPython.sh, qui fait tourner le fichier python precedant
--------------------------------


------->>>>>>>> DeNovoGenesMotifsBefore
------->>>>>>>> GenesMotifsBefore
------->>>>>>>> NcRNAMotifsBefore
------->>>>>>>> PseudogenesMotifsBefore
------->>>>>>>> RandomIntronMotifs
------->>>>>>>> RandomNonCodingMotifs


Path Motif/MotifsSearch/PollII

---------------------------------
Modif SearchMotifsParse.py
qsub RunPython.sh
--------------------------------

------->>>>>>>> DeNovoMotifsPoll2
------->>>>>>>> GeneMotifsPoll2
------->>>>>>>> IntronMotifsPoll2
------->>>>>>>> NcrnaMotifsPoll2
------->>>>>>>> NonCodingRegionMotifsPoll2
------->>>>>>>> PseudogenesMotifsPoll2


Path Motif/MotifsSearch/Regions5_3prime


------------------------------------
Modif SearchMotifsParse.py
qsub RunPython.sh
--------------------------------


------->>>>>>>> RstDeNovo5primePhylofact.txt
------->>>>>>>> 
------->>>>>>>> 
------->>>>>>>> 


*******************************************************************************************************************************
Motifs analyses
*******************************************************************************************************************************

Path: Motif/MotifsSearch/Analyse/RetrieveInfoGenes


-----------------------------------
python GetInfoDeNovo.py
--------------------------------

------->>>>>>>> DeNovoMotifsSorted
------->>>>>>>> NcRNAMotifsSorted
------->>>>>>>> PseudogenesMotifsSorted
------->>>>>>>> GenesMotifsSorted



--------------------------------
python GetInfoRandom.py
--------------------------------

------->>>>>>>> NonCodingMotifsSorted
------->>>>>>>> IntronMotifsSorted


Path: Motif/MotifsSearch/Analyse/SearchNbMotifs


--------------------------------
python MakeF1.py
--------------------------------

------->>>>>>>> DataNbGenesVsNbMotifs


--------------------------------
python MakeF2.py
--------------------------------

------->>>>>>>> DataForAverage





*******************************************************************************************************************************
Research SEG-HCA domains in de novo genes
*******************************************************************************************************************************


Path Motif/DomainAnalyse/SegHCA

-----------------------------------
python /SegHCA.py -m DeNovoGenesProt.fa
--------------------------------



*******************************************************************************************************************************
Introns BLAST
*******************************************************************************************************************************




------->>>>>>>> UCSC_Introns.tsv

Path IntronsSearch/DeNovo/



----------------------------------
python GetIntronCoordinate.py 
--------------------------------

------->>>>>>>> CoordDeNovoIntron




*******************************************************************************************************************************
Access Motifs Specific from translation in UTRs regions of de novo genes and human established genes
*******************************************************************************************************************************


Path Motif/NewTranslationMotifsUTRs


Database:

https://attract.cnic.es/download

ATtRACT database: a database of RNA binding proteins and associated motifs




--------------------------------
qsub RunProgram.sh
--------------------------------

------->>>>>>>> MotifsDistribution_3UTR.txt
------->>>>>>>> MotifsDistribution_5UTR.txt
------->>>>>>>> TotalNbMotifs_3UTR.txt
------->>>>>>>> TotalNbMotifs_5UTR.txt


*******************************************************************************************************************************
Get PTI
*******************************************************************************************************************************



--------------------------------
qsub RunProgram.sh
--------------------------------

------->>>>>>>> Data_PTI.txt















